src/components/views/rooms/MessageComposerInput.js

```diff
@@ -820,57 +820,64 @@ export default class MessageComposerInput extends React.Component {
+            if (replyingToEv) {
+                const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
+                Modal.createTrackedDialog('Emote Reply Fail', '', ErrorDialog, {
+                    title: _t("Unable to reply"),
+                    description: _t("At this time it is not possible to reply with an emote."),
+                });
+                return false;
+            }
+
             contentText = contentText.substring(4);
             // bit of a hack, but the alternative would be quite complicated
             if (contentHTML) contentHTML = contentHTML.replace(/\/me ?/, '');
-            sendHtmlFn = this.client.sendHtmlEmote;
-            sendTextFn = this.client.sendEmoteMessage;
+            sendHtmlFn = ContentHelpers.makeHtmlEmote;
+            sendTextFn = ContentHelpers.makeEmoteMessage;
         }
 
-        if (quotingEv) {
-            const cli = MatrixClientPeg.get();
-            const room = cli.getRoom(quotingEv.getRoomId());
-            const sender = room.currentState.getMember(quotingEv.getSender());
 
-            const {body/*, formatted_body*/} = quotingEv.getContent();
+        let content = contentHTML ? sendHtmlFn(contentText, contentHTML) : sendTextFn(contentText);
 
-            const perma = makeEventPermalink(quotingEv.getRoomId(), quotingEv.getId());
-            contentText = `${sender.name}:\n> ${body}\n\n${contentText}`;
-            contentHTML = `<a href="${perma}">Quote<br></a>${contentHTML}`;
+        if (replyingToEv) {
+            const replyContent = ReplyThread.makeReplyMixIn(replyingToEv);
+            content = Object.assign(replyContent, content);
 
-            // we have finished quoting, clear the quotingEvent
+            // Part of Replies fallback support - prepend the text we're sending with the text we're replying to
+            const nestedReply = ReplyThread.getNestedReplyText(replyingToEv);
+            if (nestedReply) {
+                if (content.formatted_body) {
+                    content.formatted_body = nestedReply.html + content.formatted_body;
+                }
+                content.body = nestedReply.body + content.body;
+            }
+
+            // Clear reply_to_event as we put the message into the queue
+            // if the send fails, retry will handle resending.
             dis.dispatch({
-                action: 'quote_event',
+                action: 'reply_to_event',
                 event: null,
             });
         }
 
-        let sendMessagePromise;
-        if (contentHTML) {
-            sendMessagePromise = sendHtmlFn.call(
-                this.client, this.props.room.roomId, contentText, contentHTML,
-            );
-        } else {
-            sendMessagePromise = sendTextFn.call(this.client, this.props.room.roomId, contentText);
-        }
 
-        sendMessagePromise.done((res) => {
+        this.client.sendMessage(this.props.room.roomId, content).done((res) => {
```

Comment:
Running tests locally, I found that `this.client` might not be an object at this point.